### PR TITLE
Individual Episode Cover Art

### LINKED ISSUE #255: Unique Cover Art for Episodes
### Is your enhancement related to a problem? Please describe.

I love the simplicity of this plugin, it definitely works as advertised. In the spirit of keeping it that way, I think the ability to add a unique cover art image to an individual episode would be useful as part of the core feature set. A show's cover art would alway take precedent to the episode's cover art when viewed in the podcast directories, but if the episode has its own art, this art would display on the users device when the episode is playing.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zamanq**
### Description of the Change

Added the option to select cover art for individual episodes. Right now user can use the default featured image as episode cover art and that'd also render the same metadata in the feed. With this new option they can also add an image from the Podcast Settings panel. There's a new button "Add Cover Art" which upon clicking will open the media modal. After selection user will be able to replace or remove the media as well.

So if there's no featured image selected then the new custom cover art will get used and vice-versa. 

Closes #255 

### Verification Process

Upon selecting a cover art, it'll be available in the feed channel url under item along with all the other item feed properties. The tag will be `<itunes:image href='url-of-the-image' />`

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Added - Ability to add Unique Cover Art for Episodes

### Credits

Props @zamanq @jeffpaul

### CODE REVIEW

**Comment by iamdharmesh** [comment]
Hi @zamanq, 

Thank you for the PR. 

> Right now user can use the default featured image as episode cover art and that'd also render the same metadata in the feed. 

I started reviewing the PR and am curious about the new "Add Cover Art" button. Currently, the default featured image is already being used as cover art for the episode in the feed. Have we added this button within the podcast block for better visibility and to avoid confusion, or is there another specific reason behind it?

Thanks

**Comment by zamanq** [comment]
Hi @iamdharmesh, Yes the goal was to provide an easy way for the user to set cover art individually for an episode while being on the sidebar settings for the block. The meta key `podcast_episode_cover` for this custom setting is different than the default featured image. The featured image will get priority when rendering the image in the feed and if not set then it'll look for the new custom cover art.

As a whole this seems redundant having two different options for achieving the same thing. Actually, this is due to my oversight of not inspecting the whole thing first and was really focused on creating that feature assuming that it is not introduced yet as the issue was reported in #255
cc: @jeffpaul 

Thanks!

**Comment by jeffpaul** [comment]
@iamdharmesh are you saying that if the post that the podcast is attached to has a featured image that that image gets sent as the episode image versus otherwise inheriting the default show image?

**Comment by iamdharmesh** [comment]
> @iamdharmesh are you saying that if the post that the podcast is attached to has a featured image that that image gets sent as the episode image versus otherwise inheriting the default show image?

@jeffpaul No. Currently, the plugin has already sent a featured image as an episode image (by setting `<itunes:image href='url-of-the-image' />`). However, this PR adds a new image select field on the sidebar settings for the block to make it easier for the user.

So, now we have two fields for the episode cover image:
1. Featured image
2. Episode cover image field in the block sidebar.

I would suggest checking the possibility of showing the featured image in the block sidebar, similar to what we have done for the podcast taxonomy terms (PR https://github.com/10up/simple-podcasting/pull/183). If that is possible, we can achieve this without adding additional metadata in the database. If this is not possible at the moment, then we are good with PR.

**Comment by jeffpaul** [comment]
> I would suggest checking the possibility of showing the featured image in the block sidebar, similar to what we have done for the podcast taxonomy terms (PR https://github.com/10up/simple-podcasting/pull/183). If that is possible, we can achieve this without adding additional metadata in the database. If this is not possible at the moment, then we are good with PR.

I agree, that sounds like the most ideal.  @zamanq can you research and try that if its feasible?

**Comment by zamanq** [comment]
@jeffpaul @iamdharmesh Refactored the code to leverage the post featured image as the episode cover art. If there's none, the user can select one either using the default post featured image area or using the block sidebar and it'll reflect in the featured image as well.

File: assets/js/edit.js [hunk 1]
```diff
@@ -273,6 +302,23 @@ class Edit extends Component {
...
+						<p>{__('The featured image of the current post is used as the episode cover art. Please select a featured image to set it.', 'simple-podcasting')}</p>
+						<PanelRow>
+							{featuredImageUrl && (
+								<img src={featuredImageUrl} alt="Cover Image" />
+							)}
+
+							<MediaUploadCheck>
+								<MediaUpload
+									onSelect={onUpdateImage}
+									allowedTypes={['image/gif']}
```
Comment (iamdharmesh) [inline assets/js/edit.js]:
```suggestion
									allowedTypes={['image']}
```

File: assets/js/edit.js [hunk 2]
```diff
@@ -273,6 +302,23 @@ class Edit extends Component {
...
+									render={({ open }) => (
+										<Button onClick={open}>{__('Select Cover Art', 'simple-podcasting')}</Button>
```
Comment (iamdharmesh) [inline assets/js/edit.js]: Maybe some styling is needed here? as the button gets hidden inside when we have an image available.
Screenshot 2024-02-26 at 10 44 01 PM

**Review by iamdharmesh — COMMENTED** [review]
Thanks for the changes @zamanq. Added 2 minor comments/suggestions. otherwise, looks good.

Thank you.

File: assets/js/edit.js (hunk 2 above)
Comment (zamanq) [reply assets/js/edit.js]: @iamdharmesh Added the style and while on it I introduced the Delete button for the image and updated button texts based on whether we have an image or want to add a new one as a convenience.
cc: @jeffpaul 

Screenshot 2024-02-27 at 1 35 01 PM

**Review by iamdharmesh — APPROVED** [review]
LGTM 🚀 

Thanks for working on this @zamanq.

**Pull request merged**